docs/standard/serialization/system-text-json/source-generation.md

```diff
@@ -208,6 +204,9 @@ services.AddControllers().AddJsonOptions(options =>
 
 :::zone-end
 
+> [!NOTE]
+> <xref:System.Text.Json.Serialization.JsonSourceGenerationMode.Serialization?displayProperty=nameWithType>, or fast-path serialization, isn't supported for asynchronous serialization. This limitation also applies to synchronous overloads of <xref:System.Text.Json.JsonSerializer.Serialize%2A?displayProperty=nameWithType> that accept a <xref:System.IO.Stream>.
```

Comment:
Starting with .NET 8, that's partly true. Even though streaming serialization _requires_ metadata-based models, it will fall back to fast-path assuming that the payloads are known to be small enough to fit in the predetermined buffer size. 

See https://devblogs.microsoft.com/dotnet/performance-improvements-in-net-8/#json for more details.